rpc/tests/p2p.rs

```diff
@@ -0,0 +1,273 @@
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+
+    let initial_peers = client
+        .p2p_peers()
+        .await
+        .expect("failed to get initial peer list");
+    assert!(!initial_peers.contains(&addr_info.id));
+
+    let connected_to_peer = client
+        .p2p_connectedness(&addr_info.id)
+        .await
+        .expect("failed to check initial connection to peer");
+    assert_eq!(connected_to_peer, p2p::Connectedness::NotConnected);
+
+    client
+        .p2p_connect(&addr_info)
+        .await
+        .expect("request to connect to second node failed");
+    rpc_call_delay().await;
+
+    let peers = client
+        .p2p_peers()
+        .await
+        .expect("failed to get peer list after connect request");
+    assert!(peers.contains(&addr_info.id));
+
+    let connected_to_peer = client
+        .p2p_connectedness(&addr_info.id)
+        .await
+        .expect("failed to check connection to peer after connect request");
+    assert_eq!(connected_to_peer, p2p::Connectedness::Connected);
+
+    client
+        .p2p_close_peer(&addr_info.id)
+        .await
+        .expect("Failed to close peer");
+    rpc_call_delay().await;
+
+    let final_peers = client
+        .p2p_peers()
+        .await
+        .expect("failed to get peer list after close peer request");
+    assert!(!final_peers.contains(&addr_info.id));
+
+    task.abort();
+}
+
+#[tokio::test]
+async fn protect_unprotect_test() {
+    // check whether reported protect status reacts correctly to protect/unprotect requests and
+    // whether node takes tag into the account
+
+    const PROTECT_TAG: &str = "test-tag";
+    const ANOTHER_PROTECT_TAG: &str = "test-tag-2";
+
+    let (addr_info, task) = utils::tiny_node::start_tiny_node()
+        .await
+        .expect("failed to spin up second node");
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+
+    client
+        .p2p_connect(&addr_info)
+        .await
+        .expect("request to connect to second node failed");
+    rpc_call_delay().await;
+
+    let is_protected = client
+        .p2p_is_protected(&addr_info.id, PROTECT_TAG)
+        .await
+        .expect("failed to check initial protect status");
+    assert!(!is_protected);
+
+    client
+        .p2p_protect(&addr_info.id, PROTECT_TAG)
+        .await
+        .expect("protect request failed");
+    rpc_call_delay().await;
+
+    let is_protected = client
+        .p2p_is_protected(&addr_info.id, PROTECT_TAG)
+        .await
+        .expect("failed to check protect status after protect request");
+    assert!(is_protected);
+
+    let is_protected_another_tag = client
+        .p2p_is_protected(&addr_info.id, ANOTHER_PROTECT_TAG)
+        .await
+        .expect("failed to check protect status for another tag after protect request");
+    assert!(!is_protected_another_tag);
+
+    client
+        .p2p_unprotect(&addr_info.id, PROTECT_TAG)
+        .await
+        .expect("unprotect request failed");
```

Comment:
it looks like this also should be followed by the `rpc_call_delay` as it's state changing, even tho it works there for some reason

Reply:
Connect/disconnect always required delay, so maybe this is just quick enough to usually work. Added delay just in case.